actionpack/lib/action_dispatch/middleware/exception_wrapper.rb

```diff
@@ -168,19 +173,55 @@ def clean_backtrace(*args)
         end
       end
 
+      def extract_source(trace)
+        if defined?(ErrorHighlight)
+          spot = ErrorHighlight.spot(@exception, backtrace_location: trace)
```

Comment:
What happens on versions which don't support the `backtrace_location` keyword?

Reply:
Thank you @byroot and @jhawthorn for your comments!

Because error_highlight 0.3.0 provides `ErrorHighlight.spot` but it does not support `backtrace_location` keyword, it will raise an ArgumentError: `unknown keyword: :backtrace_location`.

I hope that it should not happen because Gemfile has a requirement `gem "error_highlight", ">= 0.4.0"`, but I'm a little worried. Unfortunately we can't use `defined?` or `respond_to?` or something to determine if the new keyword is supported.

Should I prepare another API method, say, `ErrorHighlight.spot_for_backtrace_location` or something? Then we can check it by `defined?(ErrorHighlight.spot_for_backtrace_location)`. If you prefer this approach, I will gladly do so. (I wonder what the name of the method should be.)

Reply:
Thanks, that makes sense. Unfortunately we don't have the ability to control what versions users have for optional requirements, we have the Rails Gemfile, but that's really just what we use for our tests, and we have the template Gemfile, but that's only used for newly generated apps. So we do need something additional to check for this feature.

How would you feel changing the test above to

``` ruby
defined?(ErrorHighlight) && Gem::Version.new(ErrorHighlight::VERSION) >= Gem::Version.new("0.4.0")
```

Reply:
> Unfortunately we don't have the ability to control what versions users have for optional requirements

Ah, I see! I understand very well, thank you for your explanation.

> How would you feel changing the condition to

I'm fine if you are okay for this. Sorry for messing up a bit the Rails code!

Reply:
Actually, I think the preferable pattern would be:

```ruby
begin
  gem "error_highlight", ">= 0.4.0"
rescue LoadError
end
```
At the top of the file.

Reply:
@byroot Thank you! I have added the idiom at the top of `actionpack/lib/action_dispatch/middleware/exception_wrapper.rb`